File: 12_Integers_to_roman.cpp
class Solution {
public: 
   
    string intToRoman(int num) {
        string letter[]={"I","IV","V","IX","X","XL","L","XC","C","CD","D","CM","M"};
        int value[]= {1,4,5,9,10,40,50,90,100,400,500,900,1000};
        string ans;
        while(num)
        {
            for(int i=12;i>=0;i--)
            {
                int q=num/value[i];
                if(q)
                {
                    while(q--)
                        ans+=letter[i];
                    num%=value[i];
                    break;
                }
            }   
        }
        return ans;
    }
};

File: 12_Integers_to_roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        #Letters={1:'I',4:'IV',5:'V',9:'IX',10:'X',40:'XL',50:'L',90:'XC',100:'C',400:'CD',500:'D',900:'CM',1000:'M'}
        num_map = [(1000, 'M'), (900, 'CM'), (500, 'D'), (400, 'CD'), (100, 'C'), (90, 'XC'),
           (50, 'L'), (40, 'XL'), (10, 'X'), (9, 'IX'), (5, 'V'), (4, 'IV'), (1, 'I')]
        
        roman = ''

        while num > 0:
            for i, r in num_map:
                while num >= i:
                    roman += r
                    num -= i

        return roman
            